proposal.rst

```diff
@@ -168,27 +210,83 @@ Project Details
 
 **Describe any existing work and concepts on which your project is based.**
 
-FILLME
+My work will be based on the GSoC 2017 Project `Clickstream Mining for 
+Learning to Rank training data <https://trac.xapian.org/wiki/GSoC2017/LetorClickstream>`_. 
+
+**Phase 1: Implementing Click Models**
+
+The previous work has implemented the simplified DBN model for clickstream mining. 
+I plan to finish the EM algorithm in order to get a better performance.
+
+
+Besides DBN model, I will implement two more click models, the Dependent click Model 
+(DCM) and the User Browsing Model (UBM). Both two are also highly 
+effective to improve document ranking. 
+
+I will first work on an abstract class for all click models and then try to implement 
+each model according to relevant papers. 
+
+
+
+**Phase 2: Documentation**
+
+The second phase of my work will be a detailed documentation of each model I 
+implemented. It will provide the Omega users with a brief view of click models and
+pros & cons of each model. I will continue to work on the clickmodel.rst in the 
+Omega documentation files.
+
+**Phase 3: Evaluation**
+
+The third phase is about the evaluation. With more and more models applied, I 
+believe it will be necessary to provide an evaluation method for the users. I'm 
+going to use CTR estimation for the evaluation of the models. For the method 
```

Comment:
You'll need to unpack what this is — is there a paper that you'll be relying on? How are you adapting any existing methodologies to achieve what you need here? What sort of tools are you going to provide users to run an evaluation?

Reply:
I will add details to it.